File: Week02/LeetCode-0001.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        """
        O(N^2) algorithm
        """
        for i in range(len(nums)-1):
            for j in range(i+1, len(nums)):
                if nums[i] + nums[j] == target:
                    return [i, j]
        return [-1, -1]  # No such pair
        

File: Week02/LeetCode-0001.cpp
class Solution {
public:
    /// O(N^2) algorithm
    vector<int> twoSum(vector<int>& nums, int target) {
        for (int i=0; i<nums.size()-1; i++){
            for (int j=i+1; j<nums.size(); j++){
                if (nums[i] + nums[j] == target){
                    return vector<int>{i, j};
                } //end-if
            } //end-for
        } //end-for

        return {-1, -1};  // No pair adds up to the target        
    } // end-twoSum
};
